unpause/unpause.go

```diff
@@ -0,0 +1,129 @@
+package unpause
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	"github.com/jmhodges/clock"
+)
+
+const (
+	// API
+	APIVersion = "v1"
+	APIPrefix  = "/sfe/" + APIVersion
+	GetForm    = APIPrefix + "/unpause"
+
+	// JWT
+	defaultIssuer   = "WFE"
+	defaultAudience = "SFE Unpause"
+)
+
+// JWTClaims represents the claims of a JWT token issued by the WFE for
+// redemption by the SFE. The JWT contains the following claims required for
+// unpausing:
+//   - Subject: the account ID of the Subscriber,
+//   - Version: the API version this JWT was created for,
+//   - Identifiers: a set of paused identifiers.
+type JWTClaims struct {
+	jwt.Claims
+
+	// Version is the API version this JWT was created for.
+	Version string `json:"version"`
+
+	// Identifiers is set of comma separated ACME identifiers.
+	Identifiers string `json:"identifiers"`
+}
+
+// GenerateJWT generates a serialized unpause JWT with the provided claims.
+func GenerateJWT(key []byte, regID int64, identifiers []string, lifetime time.Duration, clk clock.Clock) (string, error) {
+	signer, err := jose.NewSigner(jose.SigningKey{Algorithm: jose.HS256, Key: key}, nil)
+	if err != nil {
+		return "", fmt.Errorf("creating signer: %s", err)
+	}
+
+	claims := JWTClaims{
+		Claims: jwt.Claims{
+			Issuer:    defaultIssuer,
+			Subject:   fmt.Sprintf("%d", regID),
+			Audience:  jwt.Audience{defaultAudience},
+			Expiry:    jwt.NewNumericDate(clk.Now().Add(lifetime)),
+			NotBefore: jwt.NewNumericDate(clk.Now()),
+			IssuedAt:  jwt.NewNumericDate(clk.Now()),
+		},
+		Version:     APIVersion,
+		Identifiers: strings.Join(identifiers, ","),
+	}
+
+	serialized, err := jwt.Signed(signer).Claims(&claims).Serialize()
+	if err != nil {
+		return "", fmt.Errorf("serializing JWT: %s", err)
+	}
+
+	return serialized, nil
+}
+
+// RedeemJWT deserializes an unpause JWT and returns the validated claims. The
+// key is used to validate the signature of the JWT. The version is the expected
+// API version of the JWT. This function validates that the JWT is:
+//   - well-formed,
+//   - valid for the current time (+/- 1 minute leeway),
+//   - issued by the WFE,
+//   - intended for the SFE,
+//   - contains an Account ID as the 'Subject',
+//   - subject can be parsed as a 64-bit integer,
+//   - contains a set of paused identifiers as 'Identifiers', and
+//   - contains the API the expected version as 'Version'.
+func RedeemJWT(token string, key []byte, version string, clk clock.Clock) (JWTClaims, error) {
+	parsedToken, err := jwt.ParseSigned(token, []jose.SignatureAlgorithm{jose.HS256})
+	if err != nil {
+		return JWTClaims{}, fmt.Errorf("parsing JWT: %s", err)
+	}
+
+	claims := JWTClaims{}
+	err = parsedToken.Claims(key[:], &claims)
```

Comment:
```suggestion
	err = parsedToken.Claims(key[], &claims)
```

I think reslicing the key like this shouldn't be necessary, because we already have a slice of bytes. A slice operation like that would be necessary if we had an array of bytes, `[32]byte`, which is a different type than `[]byte`. You often see this kind of reslicing with HMAC keys because they are of a known length and so are often represented as arrays.

Reply:
Fixed, this was the implementation copied from `sfe.go`.